src/Microsoft.DotNet.ProjectModel/Resolution/PackageDependencyProvider.cs

```diff
@@ -17,13 +17,15 @@ namespace Microsoft.DotNet.ProjectModel.Resolution
     public class PackageDependencyProvider
     {
         private readonly FallbackPackagePathResolver _packagePathResolver;
+        private readonly VersionFolderPathResolver _versionFolderPathResolver;
         private readonly FrameworkReferenceResolver _frameworkReferenceResolver;
 
         public PackageDependencyProvider(INuGetPathContext nugetPathContext, FrameworkReferenceResolver frameworkReferenceResolver)
         {
             if (nugetPathContext != null)
             {
                 _packagePathResolver = new FallbackPackagePathResolver(nugetPathContext);
+                _versionFolderPathResolver = new VersionFolderPathResolver(nugetPathContext.UserPackageFolder);
```

Comment:
This looks odd to me.  Why does `VersionFolderPathResolver` need the user package folder?  If it cares about that, shouldn't it care about any other "fallback" folder?